### PR TITLE
Fix #7 : Create a workflow that deploys a preview in the gh-pages

### LINKED ISSUE #7: Create a workflow that deploys a preview in the `gh-pages`
Hy if you could link the GitHub repository with Netlify or something else then contributors might find it easy to report new issues and also it will get easy for them to judge the issue and whether they can make meaningful contribution or not.

### PR DESCRIPTION
**Pull request by sid86-dev**
- Creates and deploys previews of pull requests to your GitHub Pages site
- Leaves a comment on the pull request with a link to the preview so that you and your team can collaborate on new features faster
- Updates the deployment and the comment whenever new commits are pushed to the pull request
- Cleans up after itself, removes deployed previews when the pull request is closed
- Can be configured to override any of these behaviours

### CODE REVIEW

**Comment by sid86-dev** [comment]
**This Action does not currently support deploying previews for PRs from forks**

**Comment by sid86-dev** [comment]
### Something is restricting permission.

```
remote: Permission to gdscsmit/gdscsmit.github.io.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/gdscsmit/gdscsmit.github.io.git/': The requested URL returned error: 403
```
Looks mention:

```
GITHUB_TOKEN Permissions
  Actions: read
  Checks: read
  Contents: read
  Deployments: read
  Discussions: read
  Issues: read
  Metadata: read
  Packages: read
  PullRequests: read
  RepositoryProjects: read
  SecurityEvents: read
  Statuses: read
```

Default is:

```
GITHUB_TOKEN Permissions
  Actions: write
  Checks: write
  Contents: write
  Deployments: write
  Discussions: write
  Issues: write
  Metadata: read
  Packages: write
  PullRequests: write
  RepositoryProjects: write
  SecurityEvents: write
  Statuses: write
```

See https://github.com/ad-m/cbosa/runs/3197304980?check_suite_focus=true .

**Comment by pooranjoyb** [comment]
Hey @sid86-dev 
I just needed to clarify one more thing,
- In the setup described, if multiple pull requests are opened at the same time, the subsequent PR builds will replace the previous ones in the `preview` branch.

#### This is because each new PR triggers the workflow, which builds and deploys the application to the preview branch. 

- If another PR is opened before the previous one is merged or closed, the new build will overwrite the previous one in the preview branch.

Well, this is not what we should do right?
What do you think you can do to solve this?

**Comment by pooranjoyb** [comment]
Okay @sid86-dev 
After thorough research through the setup, I feel `gh-pages` won't be that feasible for this `CI/CD` workflow.

What do you think about using Vercel? Could you write the `preview.yml` file for the same?

File: .github/workflows/greetings.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: 'Greetings'
+
+on:
+  fork:
+  push:
+    branches: [master]
```
Comment (pooranjoyb) [inline .github/workflows/greetings.yml:6]: Update this with `[main]`

**Review by pooranjoyb — APPROVED** [review]

**Comment by pooranjoyb** [comment]
Closes #74

**Pull request merged**